### PR TITLE
Add CI for macOS and Ubuntu

### PR DESCRIPTION
**Pull request by saeta**

### CODE REVIEW

**Comment by shabalind** [comment]
LGTM, I used the same configuration for swift-benchmark and it works like a charm!

File: Tests/PenguinParallelTests/NonBlockingThreadPoolTests.swift [hunk 1]
```diff
@@ -163,9 +163,9 @@ public class ThreadCountingPlatform<Underlying: ConcurrencyPlatform>: Concurrenc
     runningThreadCount += 1
     return underlying.makeThread(name: name) { [unowned self, fn] () in
       fn()
-      lock.lock()
-      runningThreadCount -= 1
-      lock.unlock()
```
Comment (dabrahams) [inline Tests/PenguinParallelTests/NonBlockingThreadPoolTests.swift:168]: Does this change belong in this PR?

Comment (saeta) [reply Tests/PenguinParallelTests/NonBlockingThreadPoolTests.swift:168]: This change was required in order to compile on mac with the latest mac toolchains.

Comment (dabrahams) [reply Tests/PenguinParallelTests/NonBlockingThreadPoolTests.swift:168]: Srsly, to *compile*?  This looks like a semantic change.

Comment (saeta) [reply Tests/PenguinParallelTests/NonBlockingThreadPoolTests.swift:168]: Yup: https://github.com/saeta/penguin/runs/660788911

**Pull request merged**